### PR TITLE
Javadoc

### PR DESCRIPTION
**Pull request by kyun99**
Add documentation for FinancialTracker.java

### CODE REVIEW

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 1]
```diff
@@ -42,7 +42,13 @@ public FinancialTracker() {
     public double calculateBalance() {
         return getTotalIncome() - getTotalExpense();
     }
-    
+
+    /**
+     * Add an Expense object into the expenses ArrayList of FinancialTracker.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:47]:
```suggestion
     * Adds an Expense object into the expenses ArrayList of FinancialTracker.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 2]
```diff
@@ -54,10 +60,16 @@ public void addExpense(Expense expense) throws ExpenseOverflowException {
         assert expenses.size() > expenseSize;
     }
 
-    private boolean isOverflowedExpense(Expense expense) throws ExpenseOverflowException {
+    private boolean isOverflowedExpense(Expense expense) {
         return expense.getValue() + getTotalExpense() > TOTAL_EXPENSE_LIMIT;
     }
 
+    /**
+     * Add an Income object into the income ArrayList of FinancialTracker.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:68]:
```suggestion
     * Adds an Income object into the income ArrayList of FinancialTracker.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 3]
```diff
@@ -69,14 +81,21 @@ public void addIncome(Income income) throws IncomeOverflowException {
         assert incomes.size() > incomeSize;
     }
 
-    private boolean isOverflowedIncome(Income income) throws IncomeOverflowException {
+    private boolean isOverflowedIncome(Income income) {
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:84]: Nice refactoring!

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 4]
```diff
@@ -69,14 +81,21 @@ public void addIncome(Income income) throws IncomeOverflowException {
...
         return income.getValue() + getTotalIncome() > TOTAL_INCOME_LIMIT;
     }
 
     private int indexOffset(int index) {
         return index - 1;
     }
 
+    /**
+     * Remove an expense entry based on its index.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:93]:
```suggestion
     * Removes an expense entry based on its index.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 5]
```diff
@@ -86,6 +105,13 @@ public Expense removeExpense(int expenseIndex) throws ExpenseEntryNotFoundExcept
         }
     }
 
+    /**
+     * Remove an income entry based on its index.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:109]:
```suggestion
     * Removes an income entry based on its index.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 6]
```diff
@@ -95,11 +121,21 @@ public Income removeIncome(int incomeIndex) throws IncomeEntryNotFoundException
         }
     }
 
+    /**
+     * Return an ArrayList called expenses from FinancialTracker.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:125]:
```suggestion
     * Returns an ArrayList called expenses from FinancialTracker.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 7]
```diff
@@ -95,11 +121,21 @@ public Income removeIncome(int incomeIndex) throws IncomeEntryNotFoundException
...
+     * 
+     * @return Return ArrayList with only expense entries inside.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:127]:
```suggestion
     * @return Returns ArrayList with only expense entries inside.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 8]
```diff
@@ -95,11 +121,21 @@ public Income removeIncome(int incomeIndex) throws IncomeEntryNotFoundException
...
+     */
     public ArrayList<Expense> getExpenses() {
         assert expenses != null;
         return expenses;
     }
 
+    /**
+     * Return an ArrayList called incomes from FinancialTracker.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:135]:
```suggestion
     * Returns an ArrayList called incomes from FinancialTracker.
```

File: src/main/java/seedu/utility/FinancialTracker.java [hunk 9]
```diff
@@ -95,11 +121,21 @@ public Income removeIncome(int incomeIndex) throws IncomeEntryNotFoundException
...
+     *
+     * @return Return ArrayList with only income entries inside.
```
Comment (AnShengLee) [inline src/main/java/seedu/utility/FinancialTracker.java:137]:
```suggestion
     * @return Returns ArrayList with only income entries inside.
```

**Review by AnShengLee — COMMENTED** [review]
Overall pretty good comments. Make sure to fix those small mistakes and typos.

File: src/main/java/seedu/utility/FinancialTracker.java (hunk 3 above)
Comment (KZQ1999) [reply src/main/java/seedu/utility/FinancialTracker.java:84]: I agree!

**Review by jonathanlkw — COMMENTED** [review]
Looks good!

File: docs/UserGuide.md [hunk 10]
```diff
@@ -94,8 +94,8 @@ to show all available commands and their format.)
 
 - Words in `UPPER_CASE` are the parameters to be supplied by you, the user.
   e.g. in add `a/AMOUNT`, `AMOUNT` is a parameter which can be typed as `a/12.30`.
-
-
+- Parameters surrounded with `[` and `]` are optional parameters which you might consider including to your input.
+  e.g. in the add expense/income feature, an optional date parameter `[D/DATE]` may be added which can be typed as `D/19/10/2021`
```
Comment (AnShengLee) [inline docs/UserGuide.md:98]: Nice add

File: docs/UserGuide.md [hunk 11]
```diff
@@ -149,11 +149,12 @@ To Terminate The Program: end
 ### Create expense entry: `add_ex`
 This command adds an expense entry to your list. Use this to keep track of your daily expenses!
 
-Format: `add_ex d/DESCRIPTION a/AMOUNT c/CATEGORY`
+Format: `add_ex d/DESCRIPTION a/AMOUNT c/CATEGORY [D/DATE]`
```
Comment (AnShengLee) [inline docs/UserGuide.md:152]: Nice add

File: docs/team/kyun99.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
+# Lim Kay Yun's Project Portfolio Page
+
+## Project: AddressBook Level 3
```
Comment (AnShengLee) [inline docs/team/kyun99.md:3]: Might want to change this later

**Review by AnShengLee — APPROVED** [review]
Overall pretty good will merge first.

**Pull request merged**